### PR TITLE
[PyTorch] Fix pad_common for float pad_value

### PR DESCRIPTION
**Pull request by shingjan**
This PR intends to fix the impl of `pad_common` for float pad_value found in models like `pytorch_unet` and `timm_efficientdet`, as well as add a couple new tests for the `torch.nn.functional.pad`.

### CODE REVIEW

File: python/tvm/relay/frontend/pytorch.py [hunk 1]
```diff
@@ -1931,16 +1931,20 @@ def pad_common(self, mode, pad_value, inputs, input_types):
         for pad in paddings:
             const_paddings.append([])
             for p in pad:
-                if not isinstance(p, int):
+                if isinstance(p, _expr.Expr):
                     p = int(_infer_value(p, {}).numpy())
+                elif isinstance(p, float):
+                    p = int(p)
+                elif not isinstance(p, int):
```
Comment (vinx13) [inline python/tvm/relay/frontend/pytorch.py:1938]: both float and int should work for `nn.pad`, casting to int here is not necessary

Comment (shingjan) [reply python/tvm/relay/frontend/pytorch.py:1938]: good catch. This used to be a fix for converting any non-`Expr` pad value, aka float, into int. `_op.nn.pad` should be able to handle both `int` and `float`

**Comment by shingjan** [comment]
@vinx13 @masahi This PR should be ready for another look. Thanks!

**Comment by shingjan** [comment]
@tvm-bot rerun

File: python/tvm/relay/frontend/pytorch.py [hunk 2]
```diff
@@ -1925,16 +1925,25 @@ def pad_common(self, mode, pad_value, inputs, input_types):
         for pad in paddings:
             const_paddings.append([])
             for p in pad:
-                if not isinstance(p, int):
+                if isinstance(p, _expr.Expr):
                     p = int(_infer_value(p, {}).numpy())
+                elif isinstance(p, float):
```
[the remark this answers is not in the document]
Comment (shingjan) [reply python/tvm/relay/frontend/pytorch.py:1930]: This is an interesting bug I am working on. When I changed this [line](https://github.com/apache/tvm/blob/f4cf7b9a8a9c926802f6e4122415c84c83fe89b5/python/tvm/relay/frontend/pytorch.py#L1908) to `padding = [pad_value] * pad_len`, the padded tensor has the correct pad_value filled in but the size is wrong. When set to `padding = [0] * pad_len`, the padded tensor has the correct size but the pad_value is filled with all zero. The root cause of this error I think is from the fact that this padding impl may have some assumptions that only work when `pad_value` is set to 0.

[the remark this answers is not in the document]
Comment (vinx13) [reply python/tvm/relay/frontend/pytorch.py:1930]: it seems the previous implementation `padding = [pad_value] * pad_len` was not correct since padding size is not related to the pad value

[the remark this answers is not in the document]
Comment (shingjan) [reply python/tvm/relay/frontend/pytorch.py]: This should be fixed. It would be great if you can take another look and see if we can merge this. Thanks!

**Review by masahi — APPROVED** [review]
